### ISSUE TITLE
NamedCurves with ffdhe

### ISSUE THREAD

**Issue by drwetter** [drwetter/testssl.sh#464]
Copy/cited from #122

--snip
On a related note, I just noticed that draft-ietf-tls-negotiated-ff-dhe extends NamedCurve in TLSv1.0 - TLSv1.2 to allow finite field Diffie-Hellman groups to be listed (as the draft of TLS v1.3 does for TLSv1.3:

```
       enum {
       // other already defined elliptic curves (see RFC 4492)
           ffdhe2048(256), ffdhe3072(257), ffdhe4096(258),
           ffdhe6144(259), ffdhe8192(260),
       //
       } NamedCurve;
```

Should I go ahead and add these to the Supported Groups (formerly Supported Elliptic Curves) extension, in the case that the client is offering any TLS_DHE_... ciphers, or wait until draft-ietf-tls-negotiated-ff-dhe becomes an RFC?
--snap

--> has been released, see https://tools.ietf.org/html/rfc7919. Can't tell who's using it yet. @dcooper16 ?

Topic for 2.9dev.

**Comment by dcooper16** [comment]
I made some changes to https://github.com/dcooper16/testssl.sh/blob/extended_tls_sockets/testssl.sh. If a server chooses a TLS_DHE_* cipher suite, then `parse_tls_serverhello()` will extract the server's ephemeral public key from the ServerKeyExchange message and compare the public key's parameters to the parameters specified in RFC 7919.

I then added code to `run_pfs()` that (for servers that support TLS_DHE_* cipher suites) would send ClientHello messages with just TLS_DHE_* cipher suites and with supported_groups extensions that specified the FFDHE groups from RFC 7919.

None of the servers that I tested used any of groups from RFC 7919, and they all seem to just ignore the presence of the supported_groups extension when using a TLS_DHE_* cipher suite.

So, perhaps we shouldn't add the FFDHE groups from RFC 7919 to the ClientHello as a general rule. But, maybe in the future we can add something like the code that I added to `run_pfs()` that will, if the server supports RFC 7919 (or TLS 1.3), list the groups from RFC 7919 that it supports.
